src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/nestedfield/NestedFieldProjection.java

```diff
@@ -88,7 +100,7 @@ private void initBoolQueryFilterIfNull() {
         return fields.stream().
                       filter(Field::isNested).
                       filter(not(Field::isReverseNested)).
-                      collect(groupingBy(Field::getNestedPath, mapping(Field::getName, toList())));
+                      collect(groupingBy(Field::getNestedPath, () -> new TreeMap<>(Collections.reverseOrder()), mapping(Field::getName, toList())));
```

Comment:
Why do we need that treemap (and an order, that it implies)?

Reply:
- Use TreeMap to generate mapping {path -> list of field domain object} in reverse alphabetical order. When we use `foreach` to traverse this map, then this order is used to generate nested query from the deepest level to the top level.
- e.g. path `project.address.city` should be nested in `project.address`, should be nested in `project`

I will add documentation for the method.

Reply:
I am still not sure why we need alphabetical order. Ideally, we should output the resulting fields in the same order as specified in the query.